hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -262,8 +280,8 @@ public static Writable avroToArrayWritable(Object value, Schema schema) {
           mapValues[1] = avroToArrayWritable(mapEntry.getValue(), schema.getValueType());
           values3[index3++] = new ArrayWritable(Writable.class, mapValues);
         }
-        wrapperWritable = new Writable[]{new ArrayWritable(Writable.class, values3)};
-        return new ArrayWritable(Writable.class, wrapperWritable);
+        // Hive 1.x will fail here, it requires values3 to be wrapped into another ArrayWritable
```

Comment:
We could, I'm not sure if Hive 2.x is backwards compatible with Hive 1.x for on-the-wire stuff. Going forward we are dropping support for Hive 1.x which means we give no guarantee that clients can compile against hive 1.x. So, a Hive 2.x client has to be able to work with Hive 1.x server (metastore/hiveserver) - I haven't tested this case since doesn't seem worth it. Let me know.

Reply:
I am fine dropping Hive 1.x